packages/compartment-mapper/src/import-parsers.js

```diff
@@ -0,0 +1,16 @@
+import parserJson from './parse-json.js';
+import parserText from './parse-text.js';
+import parserBytes from './parse-bytes.js';
+import parserCjs from './parse-cjs.js';
+import parserMjs from './parse-mjs.js';
+
+/** @import { ParserImplementation } from './types.js' */
+
+/** @type {Record<string, ParserImplementation>} */
+export const parserForLanguage = {
+  mjs: parserMjs,
+  cjs: parserCjs,
+  json: parserJson,
+  text: parserText,
+  bytes: parserBytes,
```

Comment:
in case this informs anything you're doing here: I have a branch that has exposed user-defined languages and parsers, which seems like the least-awful way to get native modules running.

Reply:
I’m not going to land this before Tuesday next week. I do not mind rebuilding this on top of your more rigorous treatment. I’m not jazzed about exposing `parserForLanguage` to the public API but it was sufficient and minimal for the bundleSource needs.

Reply:
I'm not going to land my code before Tuesday next week, so it's kind of moot.  I originally had a `parserForLanguage` option, but it turned out that I needed a new _language_, so it became an API that looks like:

```js
parsers: [
  {
    languages: ['native'], 
    parser: someParserImpl, 
    extensions: ['node']
  }
]
```

Reply:
I think I can make this work on top of your change without too much trouble. My intent is to make these `-parsers.js` modules continue to export `parserForLanguage` for internal use, but also `languageForExtension` and `parsers` like above that merges the two for public use. I can use the stub module to hide the implementation detail from the public.

Reply:
For those tuning in, we landed @boneskull’s work with some modifications to compose well with this line of work, and I removed the overlapping commits from this PR in the rebase. I then reconstructed the middle commits of this PR as #2306, as that was common ground for our separate concerns, and then removed the overlapping commits from this PR. All that remains now is the final integration in bundle-source, with very little modification from the changes that have been already approved here. The only big difference is that I’ve additionally decoupled the function that constructs a preliminary compartment map through exploration of `node_modules` on a file system.